go/pkg/cosmos/api.go

```diff
@@ -148,31 +172,33 @@ func (a *API) Account(w http.ResponseWriter, r *http.Request) {
 func (a *API) TxHistory(w http.ResponseWriter, r *http.Request) {
 	// pubkey validated by ValidatePubkey middleware
 	pubkey := mux.Vars(r)["pubkey"]
+	cursor, pageSize := a.validatePagingParams(w, r)
 
-	cursor := r.URL.Query().Get("cursor")
-
-	pageSizeQ := r.URL.Query().Get("pageSize")
-	if pageSizeQ == "" {
-		pageSizeQ = "10"
-	}
-
-	pageSize, err := strconv.Atoi(pageSizeQ)
+	txHistory, err := a.handler.GetTxHistory(pubkey, cursor, pageSize)
 	if err != nil {
-		api.HandleError(w, http.StatusBadRequest, err.Error())
+		api.HandleError(w, http.StatusInternalServerError, err.Error())
 		return
 	}
 
-	if pageSize > MAX_PAGE_SIZE_TX_HISTORY {
-		api.HandleError(w, http.StatusBadRequest, fmt.Sprintf("page size max is %d", MAX_PAGE_SIZE_TX_HISTORY))
-		return
-	}
+	api.HandleResponse(w, http.StatusOK, txHistory)
+}
 
-	if pageSize == 0 {
-		api.HandleError(w, http.StatusBadRequest, "page size cannot be 0")
-		return
-	}
+// swagger:route GET /api/v1/validators/{pubkey}/txs v1 ValidatorTxHistory
+//
+// Get paginated transaction history for a validator.
+//
+// responses:
+//
+//	200: TxHistory
+//	400: BadRequestError
+//	500: InternalServerError
+func (a *API) ValidatorTxHistory(w http.ResponseWriter, r *http.Request) {
```

Comment:
it's not the most clear, but I have been adding the validator functions to `coinstacks/cosmos/api/api.go` and `coinstacks/osmosis/api/api.go` explicitly at this time. The reason being, all cosmos-sdk coinstacks extend this base cosmos API, but not all of them support the staking module (ex. thorchain at this time). With this being declared here, the thorchain api will pick up the swagger annotations and show it as an available route that doesn't actually exist.

Until this can be composed more intelligently, can you please move this into the cosmos and osmosis coinstack api.go files directly?

Reply:
Makes sense, I moved the endpoints